clang/lib/Driver/Driver.cpp

```diff
@@ -5515,6 +5579,14 @@ class OffloadingActionBuilder final {
       delete SB;
   }
 
+  void pushForeignAction(Action *A) {
+    for (auto *SB : SpecializedBuilders) {
+      if (SB->isValid()) {
+        SB->pushForeignAction(A);
+      }
```

Comment:
small nit in regards to style, I've noticed it in a couple of places.
```suggestion
      if (SB->isValid())
        SB->pushForeignAction(A);
```